src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetMethodTests.cs

```diff
@@ -13,89 +13,34 @@ namespace System.Reflection.Tests
-        [Fact]
-        public static void TestGetSetMethod6()
+        [Theory]
+        [InlineData(typeof(ReferenceTypeHelper), "PropertyGetterSetter", true, true)]
+        [InlineData(typeof(ReferenceTypeHelper), "PropertyGetter", true, false)]
+        [InlineData(typeof(ReferenceTypeHelper), "PropertySetter", false, true)]
+        [InlineData(typeof(ReferenceTypeHelper), "Item", true, true)]
+        [InlineData(typeof(ValueTypeHelper), "PropertyGetterSetter", true, true)]
+        [InlineData(typeof(ValueTypeHelper), "PropertyGetter", true, false)]
+        [InlineData(typeof(ValueTypeHelper), "PropertySetter", false, true)]
+        [InlineData(typeof(ValueTypeHelper), "Item", true, false)]
+        [InlineData(typeof(InterfaceHelper), "PropertyGetterSetter", true, true)]
+        [InlineData(typeof(InterfaceHelper), "PropertyGetter", true, false)]
+        [InlineData(typeof(InterfaceHelper), "PropertySetter", false, true)]
+        [InlineData(typeof(InterfaceHelper), "Item", false, true)]
+        public void GetSetMethod(Type type, string propertyName, bool getter, bool setter)
         {
-            VerifyGetSetMethod(typeof(ValueTypeHelper), "PropertyGetter", true, false);
-        }
-
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod7()
-        {
-            VerifyGetSetMethod(typeof(ValueTypeHelper), "PropertySetter", false, true);
-        }
-
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod8()
-        {
-            VerifyGetSetMethod(typeof(ValueTypeHelper), "Item", true, false);
-        }
-
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod9()
-        {
-            VerifyGetSetMethod(typeof(InterfaceHelper), "PropertyGetterSetter", true, true);
-        }
-
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod10()
-        {
-            VerifyGetSetMethod(typeof(InterfaceHelper), "PropertyGetter", true, false);
-        }
+            PropertyInfo pi = getProperty(type, propertyName);
+            Assert.NotNull(pi);
 
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod11()
-        {
-            VerifyGetSetMethod(typeof(InterfaceHelper), "PropertySetter", false, true);
-        }
+            if (getter)
+            {
+                Assert.NotNull(pi.GetMethod);
+            }
 
-        //Verify GetMethod , SetMethod properties for PropertyInfo
-        [Fact]
-        public static void TestGetSetMethod12()
-        {
-            VerifyGetSetMethod(typeof(InterfaceHelper), "Item", false, true);
+            if (setter)
+            {
+                Assert.NotNull(pi.SetMethod);
+            }
```

Comment:
I realize this is what the original code was doing, but it seems like we can actually do better.  Specifically, if !getter, should we also be validating that Assert.Null(pi.GetMethod), and for !setter, that Assert.Null(pi.SetMethod)? e.g.
```C#
if (getter)
{
    Assert.NotNull(pi.GetMethod);
}
else
{
    Assert.Null(pi.GetMethod);
}
```